desc/objectives/linear_objectives.py

```diff
@@ -38,9 +38,28 @@ def update_target(self, eq):
         if self._use_jit:
             self.jit()
 
+    def _parse_target_from_user(
+        self, target_from_user, default_target, default_bounds, idx
```

Comment:
what is `idx`? It comes from below I see but I'm not sure what it is

Reply:
its the indices of which modes/etc are being constrained, in the case that not all of them are. Generally all the modes are constrained, so idx is basically just `np.arange(target.size)`